src/Couchbase.Lite.Tests.Shared/BlobTest.cs

```diff
@@ -167,5 +168,85 @@ public unsafe void TestBlobStreamCopyTo()
                 stream.Length.Should().Be(bytes.Length);
             }
         }
+
+        [Fact]
+        public void TestBlobToJSON()
+        {
+            var blob = ArrayTestBlob();
+            Action badAction = (() => blob.ToJSON());
+            badAction.Should().Throw<InvalidOperationException>(CouchbaseLiteErrorMessage.MissingDigestDueToBlobIsNotSavedToDB);
+
+            Db.SaveBlob(blob);
+
+            var json = blob.ToJSON();
+
+            var blobFromJStr = JsonConvert.DeserializeObject<Dictionary<string, object>>(json);
+            blob.JSONEquals(blobFromJStr).Should().BeTrue();
+        }
+
+        [Fact]
+        public void TestGetBlobFromDB()
+        {
+            var blob = ArrayTestBlob();
+            Db.SaveBlob(blob);
+
+            var blobDict = new Dictionary<string, object>() {
+                { Blob.ContentTypeKey, blob.ContentType },
+                { Blob.DigestKey, blob.Digest},
+                { Blob.LengthKey, blob.Length},
+                { Constants.ObjectTypeProperty, "blob" }
+            };
```

Comment:
This might be thinking a little too much but it might be good to do this via blob ToJSON instead so that if someone is looking at our tests for guidance on how to do this, they can see the supported method.  The flow that we are anticipating is:

1. Save the blob
2. Send the blob toJSON somewhere
3. Get the same JSON back later
4. Deserialize it and use it to look up the blob

Reply:
TestBlobToJSON starts from line 173 is very similar to what you describe in the steps.
Are you looking for replicator to be involved in step 2 and 3?

Reply:
No, just a start to end of the simulated flow for reference.  The replicator should not be involved here anyway, it's more of the transport between the library and third party functions.

Reply:
Added a new blob test TestDbBlobToJson to show blob ToJSON behaves correctly during transport between the library and third party functions.